web_poet/page_inputs/client.py

```diff
@@ -195,6 +196,10 @@ async def execute(
         if self.return_only_saved_responses:
             for fp, saved_data in self._saved_responses.items():
                 if request_fingerprint(request) == fp:
+                    if saved_data.exception:
+                        raise saved_data.exception
+                    if not saved_data.response:
+                        continue
```

Comment:
If there is no exception there is always going to be a response, right?

```suggestion
                    assert saved_data.response
```

Reply:
Right.

Reply:
Hey! What was the resolution here?

Reply:
Is the idea to address it later?